com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/ResourceManager.java

```diff
@@ -0,0 +1,140 @@
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package com.eclipsesource.modelserver.emf;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl;
+import org.emfjson.jackson.resource.JsonResourceFactory;
+
+import com.eclipsesource.modelserver.emf.configuration.EPackageConfiguration;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+
+/**
+ * Injectable singleton class that provides helper methods for managing
+ * (load,save...) of EMF resources.
+ *
+ */
+public class ResourceManager {
+	private static Logger LOG = Logger.getLogger(ResourceManager.class);
+	private Set<EPackageConfiguration> configurations;
+
+	@Inject
+	public ResourceManager(Set<EPackageConfiguration> configurations) {
+		this.configurations = configurations;
+		registerExtensions(configurations);
+
+	}
+
+	public void initialize() {
+		configurations.forEach(EPackageConfiguration::registerEPackage);
+	}
+
+	private void registerExtensions(Set<EPackageConfiguration> configurations) {
+		Map<String, Object> map = Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap();
+		// register default ResourceFactories (XMI and JSON)
+		map.put("*", new XMIResourceFactoryImpl());
+		map.put("json", new JsonResourceFactory(EmfJsonConverter.setupDefaultMapper()));
+		// register additional ResourceFactories
+		configurations.forEach(conf -> map.putAll(registerExtensions(conf)));
+
+	}
+
+	private Map<String, Object> registerExtensions(EPackageConfiguration configuration) {
+		Map<String, Object> map = Maps.newHashMap();
+		configuration.getFileExtensions().forEach(ext -> {
+			configuration.getResourceFactory(ext).ifPresent(fac -> map.put(ext, fac));
+		});
+
+		return map;
+	}
+
+	public Optional<Resource> loadResource(URI resourceURI, ResourceSet rs) {
+		Resource resource = rs.getResource(resourceURI, true);
+		try {
+			resource.load(Collections.EMPTY_MAP);
+			return Optional.of(resource);
+		} catch (final IOException e) {
+			LOG.error("Could not load resource with URI: " + resourceURI);
+			return Optional.empty();
+		}
+	}
+
+	public Optional<Resource> loadResource(String resourceURI, ResourceSet rs) {
+		return loadResource(URI.createURI(resourceURI), rs);
+	}
+
+	public <T extends EObject> Optional<T> loadModel(URI resourceURI, ResourceSet rs, Class<T> clazz) {
```

Comment:
Do we actually need to constraint type `T` here?

Reply:
Technically we do not need to constraint is. But types that are not derivable from EObject will always result in an empty optional so this serves as sort of a failsafe.
But we can remove the constraint if you prefer

Reply:
I see, no it's fine, let's keep it hen.